extensions/amp-next-page/1.0/visibility-observer.js

```diff
@@ -25,36 +25,65 @@ import {
 import {RelativePositions} from '../../../src/layout-rect';
 import {Services} from '../../../src/services';
 import {devAssert} from '../../../src/log';
+import {throttle} from '../../../src/utils/rate-limit';
 
 /** @enum {number} */
 export const ViewportRelativePos = {
-  INSIDE_VIEWPORT: 0,
-  OUTSIDE_VIEWPORT: 1,
-  LEAVING_VIEWPORT: 2,
-  CONTAINS_VIEWPORT: 3,
+  INSIDE_VIEWPORT: 1,
+  OUTSIDE_VIEWPORT: 2,
+  LEAVING_VIEWPORT: 3,
+  ENTERING_VIEWPORT: 4,
+  CONTAINS_VIEWPORT: 5,
 };
 
+/** @const {number} */
+const SCROLL_DIRECTION_THRESHOLD = 20;
```

Comment:
Unless I'm understanding it wrong:
```suggestion
const DELTA_TO_FLIP_SCROLL_DIRECTION = 20;
```

Reply:
It's a bit more generic than that, basically ignores all scroll events until the change is significant (and when the threshold is met it doesn't only change scroll direction but also does some other work). Renamed.